Dockerfile

```diff
@@ -0,0 +1,23 @@
+FROM golang:1.16.3-alpine as builder
+
+RUN apk add --update --no-cache make git
+
+WORKDIR /go/src/terragrunt
+
+# Download modules in a separate layer so docker caches it, to reduce build time
+# when modules have not changed.
+COPY go.mod .
+COPY go.sum .
+RUN go mod download -x
+
+COPY . .
+
+RUN make build
```

Comment:
This will end up with an empty version number in the binary. It needs to be set via:

```
-ldflags "-X main.VERSION=<VERSION>"
```

But it's not clear what version number we'd be able to use?

Reply:
The version is not empty! That's one reason I used the `make build` target. The version is determined by the command `git describe --tags --abbrev=12 --dirty --broken`. When the checkout matches a git tag, the value is just the git tag. When the checkout is not a tag, you get the last tag plus a git ref marker.

If you go with the dockerhub build service, dockerhub clones the repo and has all the tags, so this evaluates correctly. I'd expect any other container build service would also.

```
$ git describe --tags --abbrev=12 --dirty --broken
v0.29.2-2-gc90d385e21b8
$ git tag v0.29.3-testing
$ git describe --tags --abbrev=12 --dirty --broken
v0.29.3-testing
```

Reply:
For example:

```
$ docker run --rm gruntworkio/terragrunt --version
terragrunt version v0.29.2-3-gdbe93918437f-dirty
```